docs/ecosystem/typescript.md

```diff
@@ -38,27 +65,36 @@ These are the recommended `compilerOptions` for a Bun project.
     "module": "esnext",
     "target": "esnext",
     
-    // requires typescript 5.x+
-    // use "nodenext" for earlier versions
+    // if TS 5.x+
     "moduleResolution": "bundler",
+    // if TS 4.x or earlier
+    "moduleResolution": "nodenext",
 
-    // support JSX, CommonJS
-    "jsx": "react-jsx", // support JSX (value doesn't matter)
+    // support JSX, CommonJS, ES imports
+    "jsx": "react-jsx", // support JSX
     "allowJs": true, // allow importing `.js` from `.ts`
     "esModuleInterop": true, // allow default imports for CommonJS modules
+    "allowImportingTsExtensions": true,
 
     // best practices
     "strict": true,
     "forceConsistentCasingInFileNames": true,
     "skipLibCheck": true,
-
+    
     // add Bun type definitions
     "types": ["bun-types"]
   }
 }
 ```
 
-If you use `bun init`, an appropriate `tsconfig.json` is automatically generated for you.
+You'll also need to set `"type": "module"` in your `package.json` to enable ES modules.
+
+```json-diff
+  {
++   "type": "module"
+  }
+```
+
```

Comment:
This isn't true as written. `{"type": "module"}` only makes the current package an ES module, i.e. whether `*.js` files are read as ESM or CJS. It has no effect on whether ESM can be used or not since `*.mjs` files are always ESM and can be used without `{"type": "module"}`. I would recommend removing this note altogether.

Reply:
I was being reductive intentionally here to avoid getting into the weeds of TS module resolution here. But I did end up dropping this because I realized this would prevent extensionless imports in `.ts` files. It's probably better to TS4 users to avoid `type:"module"` for the time being. Thanks.

Reply:
Understood. Thanks for the edit.